### PR TITLE
panvimdoc: init at 4.0.1

### PR DESCRIPTION
**Pull request by mrcjkb**
## Description of changes

[panvimdoc](https://github.com/kdheepak/panvimdoc), a tool for generating vimdoc from markdown.
Could be used to generate vimdoc from readmes for (Neo)vim plugins that don't have a `doc` directory.

cc @teto @GaetanLepage 

- [x] Tested by running it on the nixpkgs readme.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  '';
+
+  meta = with lib; {
+    description = "Write documentation in pandoc markdown. Generate documentation in vimdoc";
+    homepage = "https://github.com/kdheepak/panvimdoc";
+    license = licenses.mit;
+    maintainers = with maintainers; [ mrcjkb ];
+    mainProgram = "panvimdoc";
+  };
```
Comment (GaetanLepage) [inline pkgs/by-name/pa/panvimdoc/package.nix:33]:
```suggestion
  meta = {
    description = "Write documentation in pandoc markdown. Generate documentation in vimdoc";
    homepage = "https://github.com/kdheepak/panvimdoc";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ mrcjkb ];
    mainProgram = "panvimdoc";
  };
```
new guidelines

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+
+  pname = "panvimdoc";
+  version = "4.0.1";
+
+  src = fetchFromGitHub {
+    owner = "kdheepak";
+    repo = "panvimdoc";
+    rev = "v${finalAttrs.version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/pa/panvimdoc/package.nix:11]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

to avoid branch/tag ambiguity?

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Minor nit-picks. Otherwise LGTM

**Comment by mrcjkb** [comment]
Result of `nixpkgs-review pr 321839` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>panvimdoc</li>
  </ul>
</details>

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Could you please format the file as well ?
`nix run .#nixfmt-rfc-style pkgs/by-name/pa/panvimdoc`

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    makeWrapper $out/share/panvimdoc.sh $out/bin/panvimdoc --prefix PATH : ${
+      lib.makeBinPath [
+        pandoc
+        vim
+        neovim-unwrapped
+      ]
+    }
+  '';
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/by-name/pa/panvimdoc/package.nix:39]:
```suggestion
  '';
  
  passthru = {
    tests.version = testers.testVersion { package = panvimdoc; };
    updateScript = nix-update-script { };
  };

  meta = {
```

Comment (mrcjkb) [reply pkgs/by-name/pa/panvimdoc/package.nix:39]: Unfortunately, the script doesn't have a `--version` or `-v` flag (and doesn't output the version).

Comment (GaetanLepage) [reply pkgs/by-name/pa/panvimdoc/package.nix:39]: Oh ok, just ignore this remark then.

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  buildPhase = ''
+    install -Dm444 scripts/* -t $out/share/scripts
+    install -Dm444 lib/* -t $out/share/lib
+    install -Dm755 panvimdoc.sh -t $out/share
+    mkdir -p $out/bin
+    makeWrapper $out/share/panvimdoc.sh $out/bin/panvimdoc --prefix PATH : ${
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:30]: this should be in postFixup

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  fetchFromGitHub,
+  stdenv,
+  pandoc,
+  vim,
+  neovim-unwrapped,
+  makeWrapper,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:12]:
```suggestion
```

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  '';
+
+  meta = {
+    description = "Write documentation in pandoc markdown. Generate documentation in vimdoc";
+    homepage = "https://github.com/kdheepak/panvimdoc";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ mrcjkb ];
+    mainProgram = "panvimdoc";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:45]: add platforms

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  src = fetchFromGitHub {
+    owner = "kdheepak";
+    repo = "panvimdoc";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-HmEBPkNELHC7Xy0v730sQWZyPPwFdIBUcELzNtrWwzQ=";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  buildPhase = ''
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:25]: add runHooks

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  version = "4.0.1";
+
+  src = fetchFromGitHub {
+    owner = "kdheepak";
+    repo = "panvimdoc";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-HmEBPkNELHC7Xy0v730sQWZyPPwFdIBUcELzNtrWwzQ=";
+  };
+
+  buildInputs = [ makeWrapper ];
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:23]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 9]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  pandoc,
+  vim,
+  neovim-unwrapped,
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:7]: why both?

File: pkgs/by-name/pa/panvimdoc/package.nix [hunk 10]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/panvimdoc/package.nix:4]: The build process use no compiler, so you can change to `stdenvNoCC`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/pa/panvimdoc/package.nix (hunk 9 above)
Comment (mrcjkb) [reply pkgs/by-name/pa/panvimdoc/package.nix:7]: https://github.com/kdheepak/panvimdoc/blob/69d9529e2a0ed53b1ba6dc56f471bf263af71042/Dockerfile#L3

**Review by GaetanLepage — APPROVED** [review]
Thanks @AndersonTorres for your detailed review !

**Review by AndersonTorres — APPROVED** [review]
Didn't test yet but lgtm.

**Comment by GaetanLepage** [comment]
I tested locally and it works fine !

**Pull request merged**